File: src/types/exercise.ts
export interface ExerciseTemplate {
  id: string;
  title: string;
  exercise_type: ExerciseType;
  primary_muscle_group: MuscleGroup;
  secondary_muscle_groups: MuscleGroup[];
  equipment: EquipmentCategory | null;
  is_custom: boolean;
}

export type ExerciseType =
  | "weight_reps"
  | "reps_only"
  | "bodyweight_reps"
  | "bodyweight_assisted_reps"
  | "duration"
  | "weight_duration"
  | "distance_duration"
  | "short_distance_weight";

export type MuscleGroup =
  | "abdominals"
  | "shoulders"
  | "biceps"
  | "triceps"
  | "forearms"
  | "quadriceps"
  | "hamstrings"
  | "calves"
  | "glutes"
  | "abductors"
  | "adductors"
  | "lats"
  | "upper_back"
  | "traps"
  | "lower_back"
  | "chest"
  | "cardio"
  | "neck"
  | "full_body"
  | "other";

export type EquipmentCategory =
  | "none"
  | "barbell"
  | "dumbbell"
  | "kettlebell"
  | "machine"
  | "plate"
  | "resistance_band"
  | "suspension"
  | "other";

File: src-tauri/src/models/exercise.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ExerciseTemplate {
    pub id: String,
    pub title: String,
    pub exercise_type: String,
    pub primary_muscle_group: String,
    pub secondary_muscle_groups: Vec<String>,
    pub equipment: Option<String>,
    pub is_custom: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct HevyExerciseTemplate {
    pub id: String,
    pub title: String,
    #[serde(rename = "type")]
    pub exercise_type: String,
    pub primary_muscle_group: String,
    pub secondary_muscle_groups: Vec<String>,
    pub is_custom: bool,
}
